### DISCUSSION TITLE
Translate works much better with word timestamps

### DISCUSSION THREAD

**Discussion by Tronic** [openai/whisper#1292] [General]
Whisper warns "Word-level timestamps on translations may not be reliable."

However, I find that the results are quite good. The only problem is that if the word order is changed in translation, the underline cannot go back but stays at the word that was spoken before. I suggest further development to allow backwards timestamps. The timing of the whole line matches speech precisely, only the underline if off in these situations.

Instead, if word level timestamps are not enabled, the whole subtitle line is badly mistimed, often by several seconds. This appears worse if preceded by a passage of silence, causing the following legends to be displayed too early. Thus it would be better to use word timestamps even in translate mode and the warning seems misguided.

**Comment by jongwook** [comment]
It's interesting that transcriptions are better with the flag! I haven't tested word-level timestamps in translation mode so I've put the warning. I guess it'd work better with the languages that have similar word ordering. Because of how it finds the alignment, it's not as straightforward to make the underline go backward.. but thanks for reporting this :)
